src/SnackSession.js

```diff
@@ -292,6 +302,99 @@ export default class SnackSession {
     return url;
   };
 
+  /**
+   * Generates a default app.json template for the snack.
+   * @returns {Object} the app.json template
+   * @function
+   */
+  generateAppJson = (): Object => {
+    const appJsonTemplate = {
+      expo: {
+        slug: this.slug,
+        name: this.name,
+        description: this.description,
+        privacy: 'unlisted',
+        sdkVersion: this.sdkVersion,
+        version: '1.0.0',
+        orientation: 'portrait',
+        primaryColor: '#cccccc',
+        icon: 'https://d1wp6m56sqw74a.cloudfront.net/~assets/c9aa1be8a6a6fe81e20c3ac4106a2ebc',
+        loading: {
+          icon: 'https://d1wp6m56sqw74a.cloudfront.net/~assets/c9aa1be8a6a6fe81e20c3ac4106a2ebc',
+          hideExponentText: false,
+        },
+        packagerOpts: {
+          assetExts: ['ttf', 'mp4', 'otf'],
+        },
+        ios: {
+          supportsTablet: true,
+        },
+        android: {
+          package: "org.example.test",
+          permissions: ["CAMERA"],
+          versionCode: 1,
+        },
+      },
+    };
+    return appJsonTemplate;
+  };
+
+  /**
+   * Builds an apk from the snack and returns a url to download the apk.
+   * @param 
+   * @returns {Promise.<string>} A promise that contains the url when fulfilled.
+   * @function
+   */
+  getApkUrlAsync = async (
+    appJson : Object
+  ): Promise<string> => {
+    const opts = {
+      platform : 'android',
+      mode: 'create',
+      isSnack: true,
+      sdkVersion: this.sdkVersion
+    };
+    let headers: any = {
+      ...(this.user.idToken ? { Authorization: `Bearer ${this.user.idToken}` } : {}),
+      ...(this.user.sessionSecret ? { 'Expo-Session': this.user.sessionSecret } : {})
+    };
+    const { id: buildId } = await buildApkAsync(appJson, headers, opts);
+
+    console.log('https://expo.io/builds/' + buildId);
+
+    const completedJob = await this.waitForApk(buildId, appJson, headers, {});
+    const artifactUrl = completedJob.artifactId
+        ? `https://expo.io/artifacts/${completedJob.artifactId}`
+        : completedJob.artifacts.url;
+
+    console.log(artifactUrl);
+    return artifactUrl;
+  };
+
+  waitForApk = async (buildId, appJson, headers, { timeout = 1200, interval = 60 } = {}) => {
+    let time = new Date().getTime();
+    await sleep(secondsToMilliseconds(interval));
+    const endTime = time + secondsToMilliseconds(timeout);
+    while (time <= endTime) {
+      const res = await buildApkAsync(appJson, headers, {
+        current: false,
+        mode: 'status',
+        platform : 'android',
+      });
+      const job = fp.compose(
```

Comment:
I have to look up compose and getOr to understand what's happening here.  i think a non-fp version would be clearer and not much more verbose